presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -286,6 +288,36 @@ public static double stLength(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
         return geometry.getEsriGeometry().calculateLength2D();
     }
 
+    @SqlNullable
+    @Description("Returns a float between 0 and 1 representing the location of the closest point on the LineString to the given Point, as a fraction of total 2d line length.")
+    @ScalarFunction("line_locate_point")
+    @SqlType(DOUBLE)
+    public static Double lineLocatePoint(@SqlType(GEOMETRY_TYPE_NAME) Slice lineSlice, @SqlType(GEOMETRY_TYPE_NAME) Slice pointSlice)
+    {
+        if (lineSlice == null || pointSlice == null) {
```

Comment:
I'm not 100% sure, but as i know if you didn't mark some parameter as nullable - the method won't ever be called with null's, and standard sql semantics for null will be applied. If it is so - this doesn't make sense. You can check for `requireNonNull` instead